iot_api/user_api/repository/NotificationPreferencesRepository.py

```diff
@@ -0,0 +1,52 @@
+import iot_logging
+LOG = iot_logging.getLogger(__name__)
+
+from sqlalchemy import not_
+
+from iot_api.user_api import db
+from iot_api.user_api.models import Tag, NotificationAssetTag
+from iot_api.user_api import Error
+from iot_api.user_api.repository import TagRepository
+
+def get_asset_tags(user_id):
+    """
+    Get the list of tags that a device must have, according to this user's
+    NotificationPreferences, in order to notify him when an alert event occurs.
+    """
+    result = db.session.query(Tag).filter(
+        NotificationAssetTag.user_id == user_id,
+        NotificationAssetTag.tag_id == Tag.id
+        ).all()
+    return result if result else []
+
+
+def upsert_asset_tags(user_id, tag_id_list):
+    """
+    Insert rows in the DB to relate this user with every tag in
+    tag_id_list, considering that they may be already related.
+    """
+    db_entries = NotificationAssetTag.find_all_by_user_id(user_id)
+    already_in_db = set()
+    for row in db_entries:
+        already_in_db.add(row.tag_id)
```

Comment:
If db_entries is a list, 29-31 could be done with 

already_in_db = set(db_entries)

Reply:
Done